src/notebooklm/cli/helpers.py

```
@@ -90,34 +90,53 @@ async def import_with_retry(
+            try:
+                existing = await client.sources.list(notebook_id)
+                existing_urls = {s.url for s in existing if s.url}
+                pending_sources = [s for s in pending_sources if s.get("url") not in existing_urls]
+                if not pending_sources:
+                    logger.info("All sources already imported; no retry needed")
+                    if not json_output:
+                        console.print("[green]All sources already imported.[/green]")
+                    return []
+            except Exception:
+                # If listing fails, retry with current pending list
+                pass
```

Ruff (0.15.10)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)